src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1892,28 +1892,51 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
 
                         id = baseId + LinesKey;
 
-                        //If we are using IInstancingRenderPackage data then we need to create a unique Geometry3D object
-                        //for each instancable item and add instance transforms.
-                        //If we have any line geometry that was not associated with an instance,
-                        //remove the previously added line data from the render package so the remaining lines can be added to the scene.
+                        //In the render package, all line geometry information (vertices, colors, indices) are stored together in the same arrays
+                        //regardless if the information is associated with an instancing transform or a line to be rendered directly to the scene.
+                        //If we are using IInstancingRenderPackage and there is vertex data associated with instancing transforms, then we need
+                        //handle the line data differently for each case.
+                        //For each instancable item, we need to create a unique Geometry3D object and add its instance transforms.
+                        //
+                        //Example: One Render package with a line, a rectangle (via instance transform) and another line
+                        //
+                        //Positions [P0,P1,P3,P3,P4,P5,P6,P7,P8] (line 1 (2 Positions), rectangle (5 Positions), line 2 (2 Positions))
+                        //Colors    [C0,C1,C2,C3,C4,C5,C6,C7,C8] (line 1 (2 Colors), rectangle (5 Colors), line 2 (2 Colors))
+                        //Indices [0,1,2,3,3,4,4,5,5,5,6,7,8] -> [(line 1) 1,2, (rectangle) 2,3,3,4,4,5,5,5,6 (line 2) 7,8]
```

Comment:
Do you mean `[(line 1) 0,1, (rectangle) 2,3,3,4,4,5,5,6 (line 2) 7,8]`?
The indices of line 1 seem incorrect and index 5 is repeated thrice instead of twice.

Reply:
YES